### DISCUSSION TITLE
Does netboot run Talos from RAM or execute an install on disk?

### DISCUSSION THREAD

**Discussion by ringods** [siderolabs/talos#7852] [Q&A] [answered]
Hello, newbie here to Talos Linux. 

My aim is to have a working cluster on a DeskPi cluster board with 5 Raspberry Pi CM4s. I'm looking into getting them provisioned via netbooting using Matchbox. In such a setup, will Talos run from RAM after netbooting, or is the OS installed on the disk? 

I would prefer the latter to prevent a netboot after possible reboots.

**Comment by smira** [comment, marked as answer]
Talos can PXE boot easily, but it will require to be installed to disk once the machine config is available.

**Comment by blackliner** [comment]
With some beefy servers having hundreds of gigabyte of memory, would it be possible to "install" talos into a ramdisk? This would fully embrace the ephemeral mindset, especially when using PXE, and leave the stateful disks to the kubernetes workloads.

**Reply by smira** [reply]
yes, certainly doable, just nobody volunteered to implement or support this feature, but there are some people asking about it

**Reply by syscod3** [reply]
i was looking to install talos via pxe in a small homelab cluster of 4 PIs and I would like to avoid putting any SDs or usb sticks into them, this or some remote mounted disk would greatly help

**Reply by smira** [reply]
it might be released eventually as part of #8367 story
